tests/libvmi/cloudinit.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package libvmi
+
+import (
+	"encoding/base64"
+
+	kvirtv1 "kubevirt.io/client-go/api/v1"
+)
+
+// WithCloudInitNoCloudUserData adds could-ini no-cloud user data.
+func WithCloudInitNoCloudUserData(data string, b64Encoding bool) Option {
```

Comment:
You can pass a function at data so you don't need the if statement, also the if statement is only needed at conversion and field.

Reply:
Please be explicit, how will you like this to look from the caller perspective?

Reply:
So I think we have two options
- Keep function firm like this but reduce the if statement scope to just the base64 encoding
- Do something like `WithCloudInitNoCloudUserData(data string, func (source *CloudInitNoCloudSource, data string))`  and from there you can creat base64 encoders and all

Reply:
The second option exposes a detail (CloudInitNoCloudSource) which is now abstracted away.
The user just wants to use the data with a specific encoding.

I am also unsure how I can implement the first, I will still need two functions due to the different members assigned  (`UserDataBase64`, `UserDataBase64`).

Reply:
Like the following: 

```golang
func(o vmiOptions) {                                                    
    o[optCloudInitNoCloudUserDataB64] = func(vmi *kvirtv1.VirtualMachineInstance) {
        diskName, bus := "disk1", "virtio"                                      
        addDiskVolumeWithCloudInitNoCloud(vmi, diskName, bus)                   
        volume := getVolume(vmi, diskName)                                      
        if b64Encoding {                                                        
            encodedData := base64.StdEncoding.EncodeToString([]byte(data))      
            volume.CloudInitNoCloud.UserDataBase64 = encodedData                
        } else {                                                                
            volume.CloudInitNoCloud.UserData = data                             
        }                                                                       
    }                                                                           
}                                           
```

Reply:
Aha, you want to inject the flag in...
I was looking to create the function so it will have no conditions internally.

As we have no performance issue and your code has fewer lines of code, I will comply.